src/config.ts

```diff
@@ -222,7 +227,42 @@ export const ALLOW_PAGOPA_IP_SOURCE_RANGE = decodeCIDRs(
 export const API_KEY = getRequiredENVVar("API_KEY");
 export const API_URL = getRequiredENVVar("API_URL");
 
-export const API_CLIENT = new ApiClientFactory(API_KEY, API_URL);
+// whether we want to reuse sockets in the fetch client
+const isFetchKeepaliveEnabled = process.env.FETCH_KEEPALIVE_ENABLED === "true";
+
+// Returns a fetch instance backed by a keepalive-enabled HTTP agent
+const getKeepaliveFetch: () => typeof fetch = () => {
+  // see https://nodejs.org/docs/latest-v10.x/api/http.html#http_new_agent_options
+  const keepAliveMaxSockets =
+    process.env.FETCH_KEEPALIVE_MAX_SOCKETS === undefined
+      ? Infinity
+      : parseInt(process.env.FETCH_KEEPALIVE_MAX_SOCKETS, 10);
+
+  // custom HTTP agent that will reuse sockets
+  const keepaliveAgent = new http.Agent({
+    keepAlive: true,
+    maxSockets: keepAliveMaxSockets
+  });
+
+  return (input, init) => {
+    const initWithKeepalive = {
+      ...(init === undefined ? {} : init),
+      agent: keepaliveAgent
+    };
+    // need to cast to any since node-fetch has a slightly different type
+    // signature that DOM's fetch
+    // TODO: possibly avoid using DOM's fetch type altoghether?
+    // tslint:disable-next-line: no-any
+    return nodeFetch(input as any, initWithKeepalive as any) as any;
+  };
+};
+
+const apiFetch: typeof fetch = isFetchKeepaliveEnabled
+  ? getKeepaliveFetch()
+  : // tslint:disable-next-line: no-any
+    (nodeFetch as any);
+
+export const API_CLIENT = new ApiClientFactory(API_KEY, API_URL, apiFetch);
```

Comment:
can you move this code into app.ts ? (this is a configuration file that should not have side effects)

Reply:
that line was already there, I've just added a parameter, I tried to keep the changes at minimum